low-level/framework/platforms/server/disk.py

```diff
@@ -0,0 +1,215 @@
+# cortx-questions@seagate.com.
+
+
+import json
+import re
+
+from cortx.utils.process import SimpleProcess
+from dbus import Array, Interface, SystemBus
+
+
+class Disk:
+
+    def __init__(self, id, path, device) -> None:
+        """Initialize Disk class."""
+        self.id = str(id)
+        self.path = str(path)
+        self.device = str(device)
+
+    def get_health(self):
+        if self._is_local_drive():
+            smartctl = "sudo smartctl"
+        else:
+            smartctl = "sudo smartctl -d scsi"
+        # Get smart availability attributes
+        cmd = f"{smartctl} -i {self.device}"
+        response, _, _ = SimpleProcess(cmd).run()
+        response = response.decode()
+        health_data = {}
+        if re.findall("SMART support is: Available - device has SMART capability", response):
+            health_data["SMART_available"] = True
+        if re.findall("SMART support is: Enabled", response):
+            health_data["SMART_support"] = "Enabled"
+        # Get smart health attributes
+        cmd = f"{smartctl} --all {self.device} --json"
+        response, _, _ = SimpleProcess(cmd).run()
+        response = json.loads(response)
+        smart_test_status = "PASSED" if response['smart_status']['passed'] else "FAILED"
+        health_data["SMART_health"] = smart_test_status
+        smart_required_attributes = set(["Reallocated_Sector_Ct", "Spin_Retry_Count", "Current_Pending_Sector", "Offline_Uncorrectable"])
+        try:
+            for attribute in response["ata_smart_attributes"]["table"]:
+                try:
+                    if attribute["name"] in smart_required_attributes:
+                        health_data[attribute["name"]] = attribute["raw"]["value"]
+                except KeyError:
+                    pass
+        except KeyError:
+            pass
+        return health_data
+
+    def _is_local_drive(self):
+        """
+        Detect Node server local drives using Hdparm tool.
+
+        Hdparm tool give information only for node drive.
+        For external JBOD/virtual drives it will give output as:
+        "SG_IO: bad/missing sense data "
+        Hdparm does not have support for NVME drives, for this drives it gives o/p as:
+        "failed: Inappropriate ioctl for device"
+        """
+        DISK_ERR_MISSING_SENSE_DATA = "SG_IO: bad/missing sense data"
+        DISK_ERR_GET_ID_FAILURE = "HDIO_GET_IDENTITY failed: Invalid argument"
+        cmd = f'sudo hdparm -i {self.device}'
+        _, err, retcode = SimpleProcess(cmd).run()
+
+        if retcode == 0:
+            return True
+        else:
+            # TODO : In case of different error(other than \
+            # "SG_IO: bad/missing sense data") for local drives,
+            # this check would fail.
+            if DISK_ERR_MISSING_SENSE_DATA not in err and \
+                DISK_ERR_GET_ID_FAILURE not in err:
+                return True
+            else:
+                return False
+
+
+class Disks:
```

Comment:
why 2nd class? function names n description can differentiate their intent, atlast resource type is same

Reply:
Disk object represent one disk( health , id etc)
Disks class creates Disk object from dbus

Reply:
class names not giving that differentiation !
why cant we merge these two classes and have only 'Disk' and let dbus related functions stay in that too

or name Disks -> something distinctive and a noun still